hoomd/hpmc/IntegratorHPMCMono.h

```diff
@@ -1072,15 +1072,18 @@ void IntegratorHPMCMono<Shape>::update(uint64_t timestep)
     \returns number of overlaps if early_exit=false, 1 if early_exit=true
 */
 template <class Shape>
-unsigned int IntegratorHPMCMono<Shape>::countOverlaps(bool early_exit)
+unsigned int IntegratorHPMCMono<Shape>::countOverlaps(bool early_exit, int query_type, bool build_tree)
     {
     unsigned int overlap_count = 0;
     unsigned int err_count = 0;
 
-    // build an up to date AABB tree
-    buildAABBTree();
-    // update the image list
-    updateImageList();
+    if (build_tree)
+        {
+        // build an up to date AABB tree
+        buildAABBTree();
+        // update the image list
+        updateImageList();
+        }
```

Comment:
```suggestion
    // build an up to date AABB tree
    buildAABBTree();
    // update the image list
    updateImageList();
```
The previous implementation was correct. `buildAABBTree` and `updateImageList` exit early when not needed.